src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -567,16 +567,20 @@ private AggregateCoercionNode GetAggregateCoercionNode(UnaryOpKind unaryOpKind,
                             continue;
                         }
 
-                        fieldCoercions.Add(
-                            fromField.Name,
-                            InjectCoercion(
-                                new ScopeAccessNode(IRContext.NotInSource(FormulaType.Build(fromField.Type)), new ScopeAccessSymbol(scope, scope.AddOrGetIndexForField(fromField.Name))),
-                                context,
-                                fromField.Type,
-                                toFieldType));
+                        var innerCoersion = InjectCoercion(new ScopeAccessNode(IRContext.NotInSource(FormulaType.Build(fromField.Type)), new ScopeAccessSymbol(scope, scope.AddOrGetIndexForField(fromField.Name))), context, fromField.Type, toFieldType);
+
+                        if (innerCoersion != null)
+                        {
+                            fieldCoercions.Add(fromField.Name, innerCoersion);
+                        }
                     }
                 }
 
+                if (!fieldCoercions.Any())
+                {
+                    return null;
```

Comment:
>return null; [](http://example.com/codeflow?start=20&length=12)

Is this safe? This method previously never returned null, so do the callers expect null?

Reply:
I verified we test the return value in all places where this is used

Reply:
Can this return child instead of null?
(and then no change needed on 599)

It's ok to change return type to IntermediateNode.

Reply:
the key is to not add a coercion node when there is no need as RecordToRecord isn't supported, hence null.